packages/@react-spectrum/dialog/src/DialogContainer.tsx

```diff
@@ -0,0 +1,86 @@
+import {DialogContainerContext} from './DialogContainerContext';
+import React, {Fragment, ReactElement, useContext, useEffect, useRef, useState} from 'react';
+import {OverlayTriggerState, useOverlayTriggerState} from '@react-stately/overlays';
+import {SpectrumDialogClose, SpectrumDialogProps} from '@react-types/dialog';
+import {unwrapDOMRef, useMediaQuery} from '@react-spectrum/utils';
+import {Modal, Popover, Tray} from '@react-spectrum/overlays';
+import {DOMRefValue} from '@react-types/shared';
+import {useOverlayPosition, useOverlayTrigger} from '@react-aria/overlays';
+import {PressResponder} from '@react-aria/interactions';
+import {DialogContext} from './context';
+import {DialogTriggerBase} from './DialogTrigger';
+import {Dialog} from './Dialog';
+
+
+export function DialogContainer(props): ReactElement {
+  let {
+    children,
+    content,
+    type = 'modal',
+    mobileType = type === 'popover' ? 'modal' : type,
+    isDismissable
+  } = props;
+  // NOTE: content needs to be saved as function in order to save in useState (since it's a function as child)
+  let [overlayContent, setOverlayContent] = useState(() => content);
```

Comment:
It's theoretically possible to have two overlays open at the same time within the same container. For example, two popovers in controlled `isOpen` state. The second one shouldn't go in its own container, because that would mean the location it is mounted would change depending on whether a previous overlay was open. Should this accept an array?